sklearn/linear_model/sag.py

```diff
@@ -0,0 +1,354 @@
+"""Solvers for Ridge and LogisticRegression using SAG algorithm"""
+
+# Authors: Danny Sullivan <dbsullivan23@gmail.com>
+#          Tom Dupre la Tour <tom.dupre-la-tour@m4x.org>
+#
+# Licence: BSD 3 clause
+
+import numpy as np
+import scipy.sparse as sp
+import warnings
+
+from ..utils import ConvergenceWarning
+from ..utils import check_array
+from ..utils.seq_dataset import ArrayDataset, CSRDataset
+from .sgd_fast import Log, SquaredLoss
+from .sag_fast import sag, get_auto_eta
+
+from .stochastic_gradient import SPARSE_INTERCEPT_DECAY
+
+
+def make_dataset(X, y, sample_weight, random_state):
+    # check which type of Sequential Dataset is needed
+    seed = random_state.randint(1, np.iinfo(np.int32).max)
```

Comment:
Please add an inline comment to explain why the `seed` should not be zero here. Please also feel free to open an issue to do an audit of the proper seeding of all occurrences of `our_rand_r` in the code base.